From dad4fcd3d063a610f09a8c7d1ae45d5e24497011 Mon Sep 17 00:00:00 2001
From: Yang Zhonghua <zhonghuax.yang@intel.com>
Date: Wed, 14 Aug 2019 13:57:09 +0800
Subject: [PATCH 1472/1548] media: ici: fix close stream, pipeline_set_power
 was called twice issue

Pipeline set power on and off should be paired.

Change-Id: I5cf0563d2f2ea4ae84072e7de5beac4f576c36ec
Tracked-On: OAM-84778
Tracked-On: OAM-84766
Tracked-On: PKT-2253
Tracked-On: HSD-1409261137
Signed-off-by: Yang, Zhonghua <zhonghuax.yang@intel.com>
---
 drivers/media/i2c/crlmodule-lite/crlmodule-core.c | 2 +-
 drivers/media/pci/intel/ici/ici-isys-stream.c     | 1 -
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/media/i2c/crlmodule-lite/crlmodule-core.c b/drivers/media/i2c/crlmodule-lite/crlmodule-core.c
index 612065625853..d6fbcb486f02 100644
--- a/drivers/media/i2c/crlmodule-lite/crlmodule-core.c
+++ b/drivers/media/i2c/crlmodule-lite/crlmodule-core.c
@@ -2206,7 +2206,7 @@ static int crlmodule_set_power(
 	dev_err(&client->dev, "crlmodule_set_power %d\n", on);
 	if (on) {
 		ret = pm_runtime_get_sync(&client->dev);
-		dev_err(&client->dev, "crlmodule_set_power val %d\n", ret);
+		dev_err(&client->dev, "%s@%d ret %d\n", __func__, __LINE__, ret);
 		if (ret < 0) {
 			pm_runtime_put(&client->dev);
 			return ret;
diff --git a/drivers/media/pci/intel/ici/ici-isys-stream.c b/drivers/media/pci/intel/ici/ici-isys-stream.c
index d2acd90dabeb..9ece9615ff26 100644
--- a/drivers/media/pci/intel/ici/ici-isys-stream.c
+++ b/drivers/media/pci/intel/ici/ici-isys-stream.c
@@ -1220,7 +1220,6 @@ static int stream_fop_release(struct inode *inode, struct file *file)
 	}
 
 	mutex_unlock(&as->isys->mutex);
-	pipeline_set_power(as, 0);
 	pm_runtime_put(&as->isys->adev->dev);
 
 	return ret;
-- 
2.20.1

